osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -221,8 +225,7 @@ protected override void Update()
                 RotationTracker.Tracking = !Result.HasResult && (OsuActionInputManager?.PressedActions.Any(x => x == OsuAction.LeftButton || x == OsuAction.RightButton) ?? false);
 
             if (spinningSample != null)
-                // todo: implement SpinnerFrequencyModulate
-                spinningSample.Frequency.Value = 0.5f + Progress;
+                spinningSample.Frequency.Value = spinnerFrequencyModulate ? 0.5f + Progress : 0.5f;
```

Comment:
Do you have a source for this formula? Wiki claims this frequency [can vary between 500 Hz and 80 kHz](https://osu.ppy.sh/help/wiki/Skinning/Sounds), although I'm not sure how accurate that is. This doesn't seem to have much to do with that range.

I also believe that `Progress` can be above 1, so this should be probably clamped from the top somehow as not to allow frequency to rise too much.

Reply:
they haven't changed the formula here, but i'm also not sure that disabling "modulation" is supposed to leave the sound at half frequency.

Reply:
I meant this comment as a "parity with stable" sort of thing. Presumably someone might complain if it's not the exact same one.

Reply:
i definitely checked when implementing. stable is (roughly) `0.5 + completion`.

but as i expected, when the skins setting is off it should *always be 1*. not 0.5

Reply:
should I still make (now two) consts for 1 and 0.5 like your review said?

Reply:
Alright, made two new consts for it and it runs fine so it should be all good now.